onnxruntime/core/providers/nv_tensorrt_rtx/nv_execution_provider.cc

```diff
@@ -1072,41 +1164,82 @@ NvExecutionProvider::~NvExecutionProvider() {
   }
 }
 
+void NvExecutionProvider::HandleCudaGraphStart(cudaStream_t stream, bool require_io_binding,
+                                               CudaGraphAnnotation_t cuda_graph_annotation_id, bool& graph_replay_on_this_run, bool& should_start_capture) {
+  graph_replay_on_this_run = false;
```

Comment:
Any reason that the CUDA graph logic and reply is being implemented inside the EP, instead of in the session? 
(see [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/inference_session.cc#L2321) and the reply [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/inference_session.cc#L2969))

Other questions are:

- CUDA graph doesn't support the case where the graph has control flow op due to it's possible that other part of the subgraph in the control flow op is not being captured and might have issue if running that subgraph during replay.
It seems there is no check here.
- Have you tried the case where the graph are partitioned and run by TRT RTX EP and CUDA EP?

Reply:
We had an internal discussion that the cuda graph logic should stay in the EP, the logic in session is the legacy approach and we will refactor it. So, let's stick to what you did to put the logic in the EP.